src/test/db.rs

```diff
@@ -318,3 +323,61 @@ async fn db_aggregate_disk_use() {
         .await
         .expect("aggregate with disk use should succeed");
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn create_index_options_defaults() {
+    let defaults = IndexOptionDefaults {
+        storage_engine: doc! { "wiredTiger": doc! {} },
+    };
+    index_option_defaults_test(Some(defaults)).await;
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+async fn create_index_options_defaults_not_specified() {
+    index_option_defaults_test(None).await;
+}
+
+#[function_name::named]
+async fn index_option_defaults_test(defaults: Option<IndexOptionDefaults>) {
+    let _guard = LOCK.run_concurrently().await;
+
+    let mut options = CLIENT_OPTIONS.clone();
+    if TestClient::new().await.is_sharded() {
+        options.hosts = options.hosts.iter().cloned().take(1).collect();
+    }
+    let client = EventClient::with_options(options).await;
+    let db = client.database(function_name!());
+
+    let options = CreateCollectionOptions::builder()
+        .index_option_defaults(defaults.clone())
+        .build();
+    db.create_collection(function_name!(), options)
+        .await
+        .unwrap();
+    db.drop(None).await.unwrap();
+
+    let events = client.command_events.read().unwrap();
+    let mut iter = events.iter().filter(|event| match event {
+        CommandEvent::CommandStartedEvent(CommandStartedEvent { command_name, .. }) => {
```

Comment:
Rather than doing the pattern match twice, I think using `https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map` above to return the command would be a little cleaner.